### ISSUE TITLE
Interview Question(s) for Deep Learning Positions

### ISSUE THREAD

**Issue by sahiljuneja** [udacity/dlnd-issue-reports#263]
This is based on a discussion Brok and I had on the Slack and he asked me to create this post. We know this ND doesn't offer career services, but this can perhaps help you guys to structure the course with content that can help students directly for job applications.

I interviewed with a company recently for a Deep Learning engineer position. The position had some of the usual (minimum) requirements specific to the self-driving car domain, such as -

- M.S. in related field.
- Experience with CNNs for applications in object detection, image segmentations, sensor fusion etc.
- Experience with some Deep Learning Frameworks like Caffe, Tensorflow etc.
- Fluency in C++ or Python

They asked me one question on Deep Learning, which as per them was a basic question everyone should be familiar with -

Given a specific CNN model (64x64 input image sizes, 1 layer with 3x3x16 filter, SAME padding with stride 1, max pooling with stride 2, and a FC layer with 100 width for example) what would be its memory footprint? Related question https://www.quora.com/How-do-you-calculate-the-memory-footprint-of-a-particular-deep-learning-model

My answer was based on calculating the total number of parameters for that model based on the filter size, size of inputs, and everything else, related to size of the variables and calculate a rough estimate on the memory usage. But I lacked in explaining my thought process clearly and perhaps fumbled enough to give them the idea I wasn't sure of this at all and they didn't proceed.

From what I gather, they expected me to break down the model properly and explain it step-by-step and also include details on backpropagation while calculating the footprint. They commented that this is something that everyone should be able to do so I believe it's an important aspect to learn - breaking down the model clearly as well. 

I hope the above helps you guys in some way and fellow students as well.

Some questions for other ongoing interviews I have had recently have focused on -
- What are dropouts and how do they help?
- What are activation functions and which is the most popular one and why?
- How did you manage to reduce the overall size of this model in this specific project?

[I'll try and update the above list if you think it can help]
